src/components/pages/Home.tsx

```diff
@@ -38,34 +29,57 @@ const queryLatest = {
 function SectionQueryResult({
   title,
   query,
+  queryType,
   action
 }: {
   title: ReactElement | string
-  query: SearchQuery
+  query?: SearchQuery
+  queryType?: string
   action?: ReactElement
 }) {
   const { config } = useOcean()
   const [result, setResult] = useState<QueryResult>()
+  const [loading, setLoading] = useState(true)
 
   useEffect(() => {
     if (!config?.metadataCacheUri) return
-
     const source = axios.CancelToken.source()
 
     async function init() {
-      const result = await queryMetadata(
-        query,
-        config.metadataCacheUri,
-        source.token
-      )
-      setResult(result)
+      setLoading(true)
+      try {
+        if (queryType && queryType === 'graph') {
+          const ddoList: DDO[] = await getHighestLiquidityAssets(
```

Comment:
so in the end as requested we'll simply do a `queryMetadata`, only with a modified `query` passed to it. This would imply we do not need this if/else logic here at all, and rather have a helper method to construct the query and then pass it to `< SectionQueryResult  />` component further down. Then also no need for any new props like `queryType`